src/main/java/Deadline.java

```diff
@@ -0,0 +1,46 @@
+public class Deadline extends Task {
+    protected String by;
+
+    public Deadline(String description, String by) {
+        //parent class sets String description and bool isDone
+        super(description);
+        this.by = by;
+        this.type = "D";
+    }
+
+    @Override
+    public String toString() {
+        return "[D]" + super.toString() + " (by: " + by + ")";
+    }
+
+    public static Deadline extractTaskAndDueDate (String input) {
+        Deadline taskAndDeadlineString = new Deadline("random","random");
+        //splits string according to /by keyword
+        String[] parts = input.split("/by");
+        //extracts task portion from input, after deadline keyword
+        taskAndDeadlineString.description = parts[0].substring("deadline".length()).trim();
+        //extracts deadline portion from input
+        taskAndDeadlineString.by = parts[1].trim();
+        return taskAndDeadlineString;
+    }
+
+    public String getBy() {
+        return by;
+    }
+
+    public static void printDeadline (Deadline description) {
+        Omoh.printHorizontalLine();
+        System.out.println("Got it. I've added this task:");
+        System.out.println("[D][ ] " + description.description + " (by: " + description.by + ")" );
```

Comment:
Perhaps the .type variable can be used here to replace "[D]", allowing for easier change of type letters in the future.

Reply:
Oh yes thats a good idea. Thank you for the insight!